src/compas/datastructures/network/core/graph.py

```diff
@@ -65,15 +65,22 @@ def DATASCHEMA(self):
     def JSONSCHEMANAME(self):
         return 'graph'
 
-    def __init__(self):
-        super(Graph, self).__init__()
+    def __init__(self,
+                 name=None,
+                 default_node_attributes=None,
+                 default_edge_attributes=None):
+        super(Graph, self).__init__(name=name)
         self._max_node = -1
-        self.attributes = {'name': 'Graph'}
+        self.attributes = {}
         self.node = {}
         self.edge = {}
         self.adjacency = {}
-        self.default_node_attributes = {}
+        self.default_node_attributes = {'x': 0.0, 'y': 0.0, 'z': 0.0}
```

Comment:
I'm surprised to see this.  I thought `Graph`s were purely topological.

Reply:
they are. before this was added in the network init. however, even graphs have to be visualised so purely from a practical point of view we might as well add the default coordinates here. can change this back if it doesn't feel right ...

Reply:
in general, i think the separation between graph and network, and between halfedge and mesh, etc. is a bit artificial and is only there to make the individual class definitions a bit shorter. especially in case of the mesh it doesn't make much sense because halfedge is not meaningful by itself...

Reply:
If a graph were visualized with all the nodes at the origin, it's not much of a visualization...

Reply:
indeed it is not. the defaults are just there such that more meaningful coordinates can be set through the attribute functions. if no geometry is explicitly assigned, the graph could be considered as purely topological and visualisations are meaningless without a chosen layout. we could wrap the layout mechanisms of `networkx` for this...

Reply:
I would vote for not having coords in the graph and keeping it as purely topological as possible. Then, as mentioned, visualization needs layouting, but we can either use some of the `networkx` ones, or our own, I wrote a few simple ones for our workshops, which surely need improvement but might be a starting point.

Reply:
sure will move it to network